libraries/AP_HAL/EventHandle.cpp

```diff
@@ -0,0 +1,33 @@
+#include "EventHandle.h"
```

Comment:
So the implementation goes like this, there can be different types of Sources, but EventHandle is always AP_HAL::EventHandle. Since there are wide varieties of EventSources, It's better that we allow for implementation of EventSource as needed. EventSource is something that is hidden to the user and stays bellow HAL, or even Library that intends to use this. Also, all types of EventSources have to be derived from AP_HAL::EventSource.

In the case of CAN, in ChibiOS we create a common EventSource, there is no need for it to be HAL_Driver specific. But in the case of Linux, pollfd method used by CANIface is very CAN specific, so in that case we create CANIface::CANSocketEventSource which inherits AP_HAL::EventSource. But to the external user in this case CANProtocolDrivers, all they need is to pass the EventHandle to the HAL_Driver, so it can bind it with Source, and CANProtocolDrivers can be totally oblivious to what goes on in EventSource. Hence I believe this method gives us the flexibility to implement for wide variety of scenarios, at the same time ensures uniform use.